src/components/panel/base-panel-header.tsx

```diff
@@ -0,0 +1,23 @@
+import { Color } from "@lifesg/react-design-system/color";
+import { Text } from "@lifesg/react-design-system/text";
+import { ChevronLeftIcon } from "@lifesg/react-icons/chevron-left";
+import { ChevronRightIcon } from "@lifesg/react-icons/chevron-right";
+import { useBuilder } from "../../context-providers/builder/hook";
+import { IconButton } from "../common/icon-button";
+import { BasePanelHeaderWrapper } from "./base-panel.styles";
+
+export const BasePanelHeader = () => {
+    const { state, togglePanel } = useBuilder();
+    return (
+        <BasePanelHeaderWrapper>
+            <IconButton
+                $iconSize="1.5rem"
+                $iconColor={Color.Neutral[3]}
+                onClick={() => togglePanel(!state.showPanel)}
+            >
+                {state.showPanel ? <ChevronLeftIcon /> : <ChevronRightIcon />}
```

Comment:
a nice touch would be to rotate it as it change. But i'll accept it as this for now.